include/circt/Support/LoweringOptions.h

```diff
@@ -62,6 +62,10 @@ struct LoweringOptions {
   // Otherwise, it will print them as always statements
   bool useAlwaysFF = false;
 
+  // If true, ExportVerilog emits AlwaysCombOp as Verilog always_comb
+  // statements.  Otherwise, it will print them as `always @(*)`.
+  bool useAlwaysComb = false;
```

Comment:
Thank you for adding this patch, but please invert the sense of this.  We should default to generating "best quality" "most recent features" and then allow people to down configure from there.

I think we should flip `useAlwaysFF` into a `disallowAlwaysFF` for similar reasons.  WDYT?

Reply:
I think pushing to use the newest features is the right way to go, but things are a little bit hairy here around `always_ff`.

When talking about this with the group today, we discussed how `always_comb` has some differences from `always @(*)` and that they are not totally equivalent. We might have been playing it too loose with these lowering options only affecting ExportVerilog, and we may need to interpret the lowering flags across a few passes to generate the correct kind of `sv.always` op up-front. I do think we can turn on `always_comb` by default without a problem right now though.

We [originally disabled](https://github.com/llvm/circt/pull/1310) `always_ff` generation because the way we were using it was illegal in some tools:  you can't drive a value in both an `initial` block and an `always_ff` block. At the time I thought it might be a badly behaving tool, but it turns out this is the correct way to interpret the spec. This is a really unfortunate limitation, we might be able to get around it by changing how initialization works.

I don't think we have a solid plan of action yet.  I'm a little hesitant to turn on `always_ff` when we know the generated verilog is broken. I could enable the `always_ff` flag but only generate regular `always` statements when lowering from FIRRTL.

Reply:
Ok, I agree that always_ff sounds like a different problem if that is the case.  It sounds like our solution there is just completely incorrect.  I'll file a separate bug about this.  Can you take care of flipping the sense of always_comb?

Reply:
Will do!

Reply:
Just a reminder: `always @(*)` and `always_comb` has differences in it's behavior at time 0.

Reply:
update to always_comb printing here: https://github.com/llvm/circt/pull/1709